src/main/java/org/spongepowered/api/util/weighted/WeightedCollection.java

```diff
@@ -56,22 +57,19 @@ public WeightedCollection() {
     /**
      * Selects a random value from this list based on their weight.
      * 
-     * <p>If the list is empty then null will be returned.</p>
-     * 
      * @param rand The random object to use for selection
-     * @return The selected value, or null if the list was empty
+     * @return The selected value, if the list is not empty
      */
-    public T get(Random rand) {
+    public Optional<T> get(Random rand) {
```

Comment:
I'd rather not really, to be more in the merit of the standard collection types. If you don't want to return null then throw an exception instead, but don't make it Optional.